### PR TITLE
Remove old export code, and fix permissions

### PR DESCRIPTION
**Pull request by dannyroberts**
The two things are only topically related, but I figured I'd throw them in the same PR.

I noticed when I was looking to delete old export code that you can no longer give users granular permissions for exports, which is not a big deal except for De-ID exports, where that's pretty crucial. (There's less point in de-identifying exports if to get access to them you have to also have access to all full exports.)

### CODE REVIEW

File: corehq/apps/reports/util.py [hunk 1]
```diff
@@ -346,6 +347,10 @@ def get_possible_reports(domain_name):
                     'path': model.__module__ + '.' + model.__name__,
                     'name': model.name
                 })
+
+    for slug, name, is_visible in get_extra_permissions():
+        if is_visible(domain):
+            reports.append({'path': slug, 'name': name})
```
Comment (dannyroberts) [inline corehq/apps/reports/util.py]: This is the main permissions change. By adding this to `get_possible_reports`, the 3 exports permissions will display to the user in the per-report permissions section. Behind the scenes, they're still being saved with the same path as before, even though there's no longer a report at that path: we're just treating it as a string constant—it never really mattered that it corresponded to the path, that just set up a easy mapping when I first wrote it. (And yes, I regret writing it that way back in the day.)

**Comment by snopoke** [comment]
Nice deletion ratio. Build fail seems related:
```
======================================================================
FAIL: testExportFilter (corehq.apps.reports.tests.test_export_api.ExportTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/reports/tests/test_export_api.py", line 140, in testExportFilter
    self.assertEqual(302, get_export_response(c).status_code)
AssertionError: 302 != 404
```

**Comment by dannyroberts** [comment]
Thanks, just pushed a fix to the tests. @czue I'm pretty sure this is fine; I'm changing export API's 302s to 404s. They're APIs so the 302s didn't really make sense. I made the change because

1. the url is never hit with ?next=<something> as far as I can tell from the nginx logs
2. it was defaulting to redirecting the old export page which has been 404ing since the url entry was removed.

**Comment by dannyroberts** [comment]
@biyeun cc: @nickpell

**Comment by dannyroberts** [comment]
:white_check_mark:

**Comment by biyeun** [comment]
:heart:

**Comment by dannyroberts** [comment]
:goberserk:

**Comment by czue** [comment]
whee! ratio is even better if you look at tests/non-tests. :skull:

**Pull request merged**